bfsdk/cmake/CMakeGlobal_Project.txt

```diff
@@ -16,147 +16,33 @@
 # License along with this library; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
 
-# ------------------------------------------------------------------------------
-# Includes
-# ------------------------------------------------------------------------------
-
-if(NOT DISABLE_CTEST)
-    include(CTest)
-endif()
-
 include(ExternalProject)
+include(${CMAKE_CURRENT_LIST_DIR}/CMakeOption_Coverage.txt)
+include(${CMAKE_CURRENT_LIST_DIR}/CMakeOption_USan.txt)
+include(${CMAKE_CURRENT_LIST_DIR}/CMakeOption_ASan.txt)
+include(${CMAKE_CURRENT_LIST_DIR}/CMakeTarget_Tidy.txt)
+include(${CMAKE_CURRENT_LIST_DIR}/CMakeTarget_Format.txt)
 
-# ------------------------------------------------------------------------------
-# Packages
-# ------------------------------------------------------------------------------
-
-find_package(Git REQUIRED)
-
-# ------------------------------------------------------------------------------
-# Source / Binary Dirs
-# ------------------------------------------------------------------------------
-
-if(BAREFLANK_SOURCE_DIR)
-    message(STATUS "bareflank src dir: ${BAREFLANK_SOURCE_DIR}")
-endif()
-
-if(BAREFLANK_BINARY_DIR)
-    message(STATUS "bareflank bin dir: ${BAREFLANK_BINARY_DIR}")
-endif()
-
-# ------------------------------------------------------------------------------
-# CI Options
-# ------------------------------------------------------------------------------
-
-option(ENABLE_COVERAGE "enable code coverage" OFF)
-option(ENABLE_DYNAMIC_ASAN "enable google address sanitizer" OFF)
-option(ENABLE_DYNAMIC_USAN "enable google undefined sanitizer" OFF)
-
-if(ENABLE_COVERAGE OR ENABLE_DYNAMIC_ASAN OR ENABLE_DYNAMIC_USAN)
-    set(ENABLE_UNITTESTING ON)
-endif()
-
-# ------------------------------------------------------------------------------
-# Developer Mode Options
-# ------------------------------------------------------------------------------
-
-option(ENABLE_TIDY "enable clang tidy" OFF)
-option(ENABLE_ASTYLE "enable astyle checks" OFF)
-option(ENABLE_UNITTESTING "enable unit testing" OFF)
-option(ENABLE_DEVELOPER_MODE "enable debug mode, astyle and clang tidy" OFF)
-
-if(ENABLE_DEVELOPER_MODE AND NOT WIN32)
-    set(ENABLE_TIDY ON)
-    set(ENABLE_ASTYLE ON)
-    set(UPDATE_DISCONNECTED ON)
+if(NOT CMAKE_CXX_SIZEOF_DATA_PTR STREQUAL "8")
+    message(FATAL_ERROR "Only 64bit builds are supported")
 endif()
 
-if(NOT CMAKE_BUILD_TYPE)
-    if(ENABLE_UNITTESTING OR ENABLE_DEVELOPER_MODE)
-        set(CMAKE_BUILD_TYPE "Debug")
-    else()
-        set(CMAKE_BUILD_TYPE "Release")
-    endif()
+if(ENABLE_UNITTESTING)
+    include(CTest)
```

Comment:
huh.... this might work. I would make sure that unit tests all run and pass based on both static and shared builds, but if this works, this is cleaner for sure.